### DISCUSSION TITLE
TS issues around auth.user object (possibly undefined)

### DISCUSSION THREAD

**Discussion by parker-codes** [adonisjs/core#1380] [Help] [answered]
When I destructure the auth object in a controller and attempt to access the `user` property, TS shows me that it is possibly undefined. This route is protected by an auth middleware though, so it is not undefined.

My issue is that every time I need to access something, I use the chaining operator, like `auth.user?.some_field`, but that means in order for my regular typing to work, I need to specify `as string` after it or some other forceful typing. It's cluttering up the code a bit.

I ended up coercing a user variable with the model class to use throughout, but I'm hoping this could be built-in somehow.
```ts
const user = auth.user as User;
```

I know it would be difficult to detect that a controller is behind an auth middleware, so I guess this typing would need to either be enforced by an in-controller switch or some other TS strategy.

**Comment by RomainLanz** [comment, marked as answer]
Hey @parker-codes! 👋 

TypeScript offers the `!` syntax that force the variable to be defined.

```ts
         👇🏼
auth.user!.some_field
```

**Reply by parker-codes** [reply]
That's a great solution! I still think it would be nice to have it somehow know that it's behind auth as I mentioned, but I do like this solution. Instead of specifying the "non-null assertion" (`!`) on every instance, I just do:

```ts
const user = auth.user!;
```

**Reply by thetutlage** [reply]
Suggestions are welcome, if you think it can be improved
